src/coreComponents/common/format/table/TableData.hpp

```diff
@@ -149,17 +163,31 @@ class TableData2D
   std::set< real64 > m_columnValues;
 };
 
+template< typename T >
+constexpr bool isCellType = std::is_same_v< T, CellType >;
+
 template< typename ... Args >
 void TableData::addRow( Args const &... args )
 {
-  std::vector< string > m_cellsValue;
+  std::vector< CellData > cells;
   ( [&] {
-    static_assert( has_formatter_v< decltype(args) >, "Argument passed in addRow cannot be converted to string" );
-    string const cellValue = GEOS_FMT( "{}", args );
-    m_cellsValue.push_back( cellValue );
+    static_assert( has_formatter_v< decltype(args) > || isCellType< std::decay_t< decltype(args) > >, "Argument passed in addRow cannot be converted to string nor a CellType" );
+    if constexpr (std::is_same_v< Args, CellType >) {
+      if( args == CellType::Separator )
+      {
+        cells.push_back( {CellType::Separator, "-"} );
+      }
+      else
+      {
+        cells.push_back( {CellType::MergeNext, " "} );
+      }
```

Comment:
I would have expected something simpler here (with filled `CellData::type`, and empty `CellData::value`):
```suggestion
      cells.push_back( CellValue( args ) );
```

I don't really see why you need those special values here,
1. For `CellType::Separator`, `"-"` will, I think, almost always expected to be the `m_horizontalLine`. If you want some customization, please default to `m_horizontalLine` when an empty entry is detected **on the `TableFormatter` side!** Or you can just always consider `m_horizontalLine` without reading `CellData::value` (which should be empty anyway).
2. For `CellType::MergeNext`, `" "` is always meant to be the space character, as the space is always the *"empty space character"*.

Anyway, there is no obligations to fill strings as they default to an empty string (= **no allocation**).